src/lib.rs

```diff
@@ -257,19 +257,28 @@ pub trait Segment: Sized + Debug {
 pub enum SharedLibraryId {
     /// A UUID (used on mac)
     Uuid([u8; 16]),
+    /// A GNU build ID
+    GnuBuildId(Vec<u8>),
+}
+
+impl SharedLibraryId {
+    /// Returns the raw bytes of the shared library ID.
+    pub fn as_bytes(&self) -> &[u8] {
+        match *self {
+            SharedLibraryId::Uuid(ref bytes) => &*bytes,
+            SharedLibraryId::GnuBuildId(ref bytes) => &bytes,
+        }
+    }
 }
 
 impl fmt::Display for SharedLibraryId {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            SharedLibraryId::Uuid(ref bytes) => {
-                for (idx, byte) in bytes.iter().enumerate() {
-                    if idx == 4 || idx == 6 || idx == 8 || idx == 10 {
-                        try!(write!(f, "-"));
-                    }
-                    try!(write!(f, "{:02x}", byte));
-                }
+        let bytes = self.as_bytes();
```

Comment:
It seems strange to format the `GnuBuildId` as a `Uuid` when it isn't a uuid.

Reply:
This is the format that lldb uses, but we can change it.

Reply:
Ah okay, not sure it matters much.

Reply:
changed